src/Sarif.Sarifer/BackgroundAnalysisTextViewCreationListener.cs

```diff
@@ -71,15 +74,80 @@ public void TextViewCreated(ITextView textView)
 
             textView.Closed += (object sender, EventArgs e) => this.TextView_Closed(textView);
 
+            // TextBuffer.Changed event is a performance critical event, whose handlers directly affect typing responsiveness.
+            // Unless it's required to handle this event synchronously on the UI thread.
+            // As suggested listen to Microsoft.VisualStudio.Text.ITextBuffer2.ChangedOnBackground event instead.
+            if (textView.TextBuffer is VisualStudio.Text.ITextBuffer2 oldBuffer)
+            {
+                oldBuffer.ChangedOnBackground += (object sender, VisualStudio.Text.TextContentChangedEventArgs e) => this.TextBuffer_ChangedOnBackground(textView);
+            }
+
             this.textBufferViewTracker.AddTextView(textView, path, text);
+
+            this.inputAssistant = new TextEditIdleAssistant();
+            this.inputAssistant.Idled += this.InputAssistant_Idled;
+        }
+
+        public void Dispose()
+        {
+            this.Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!this.disposed)
+            {
+                if (disposing)
+                {
+                    this.textBufferViewTracker.Clear();
+                    this.inputAssistant.Dispose();
+                }
+
+                this.disposed = true;
+            }
+        }
+
+        private void InputAssistant_Idled(object sender, TextEditIdledEventArgs e)
+        {
+            ITextView textView = e.TextView;
+            string text = textView.TextBuffer.CurrentSnapshot.GetText();
+            this.textBufferViewTracker.UpdateTextView(textView, string.Empty, text);
+        }
+
+        private void TextBufferViewTracker_ViewUpdated(object sender, ViewUpdatedEventArgs e)
+        {
+            this.BackGroundAnalyzeAsync(e.Path, e.Text, e.CancellationToken)
+                .FileAndForget(FileAndForgetEventName.BackgroundAnalysisFailure);
+        }
+
+        private void TextBuffer_ChangedOnBackground(ITextView textView)
+        {
+            this.inputAssistant.TextChanged(new TextEditIdledEventArgs(textView));
         }
 
         private void TextBufferViewTracker_FirstViewAdded(object sender, FirstViewAddedEventArgs e)
         {
-            this.backgroundAnalysisService.Value.AnalyzeAsync(e.Path, e.Text, e.CancellationToken)
+            this.BackGroundAnalyzeAsync(e.Path, e.Text, e.CancellationToken)
                 .FileAndForget(FileAndForgetEventName.BackgroundAnalysisFailure);
         }
 
+        private async System.Threading.Tasks.Task BackGroundAnalyzeAsync(string path, string text, CancellationToken cancellationToken)
```

Comment:
>BackGround [](start = 50, length = 10)

Background

Reply:
fixed

---
In reply to: [564964548](https://github.com/microsoft/sarif-visualstudio-extension/pull/349#discussion_r564964548) [](ancestors = 564964548)